Assets/Scripts/Legacy/RCextensions.cs

```diff
@@ -171,6 +172,27 @@ public static float returnFloatFromObject(object obj)
         return 0f;
     }
 
+    public static T SafeGet<T, T1, T2>(this IDictionary<T1, T2> h, T1 key) where T : new()
```

Comment:
These extension methods seem fine, but:
1. Move it to a dedicated extensions class, as RCExtensions is declared obsolete so no new logic should be added here
2. For extension methods, please also include /// summary comments